app/job_creator.rb

```diff
@@ -20,4 +17,10 @@ def create_job
 
     puts "Job: #{@job_data[:title]} saved to the database."
   end
+
+  private
+
+  def find_job
+    existing_job = Job.find_by(url: @job_data[:url])
```

Comment:
Memoization is a technique used **to cache the result of a method call** so that subsequent calls with the same arguments return the result from the cache instead of recalculating it.

In Ruby, memoization is commonly implemented using the ||= operator, which assigns a value to a variable if the variable is either nil or not defined. Here's how you can apply memoization to the find_job method:

```
def find_job
  @existing_job ||= Job.find_by(url: @job_data[:url])
end
```

Reply:
Used memoization, also used rubocop formating.